### PR TITLE
Added localization to the biomes.

### PR DESCRIPTION
**Pull request by CodeTappert**
German Localization by me. (Native German).
French Localiztation by sangara42 from Discord

I know that there is already pr that includes this. But as Benny says it should be split up so that i am doing here.

It working for Map and Biome Identifier:

image

It working in summary
image

### CODE REVIEW

**Comment by CodeTappert** [comment]
Btw this is the bloated pr where most if not all of this is from (besides the french translation):
https://github.com/pagefaultgames/pokerogue/pull/918

File: src/locales/de/biome.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  "SPACE": "Stratosphäre",
+  "CONSTRUCTION_SITE": "Baustelle",
+  "JUNGLE": "Dschungel",
+  "FAIRY_CAVE": "Feenhöhle",
+  "TEMPLE": "Tempel",
+  "SLUM": "Elendsviertel",
+  "SNOWY_FOREST": "Verschneiter Wald",
+  "ISLAND": "Insel",
+  "LABORATORY": "Labor",
+  "END": "Ende",
```
Comment (bennybroseph) [inline src/locales/de/biome.ts:39]: We set this to ??? to keep mystery around what this biome actually is in the lore.

File: src/locales/en/biome.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  "SPACE": "Stratosphere",
+  "CONSTRUCTION_SITE": "Construction Site",
+  "JUNGLE": "Jungle",
+  "FAIRY_CAVE": "Fairy Cave",
+  "TEMPLE": "Temple",
+  "SLUM": "Slum",
+  "SNOWY_FOREST": "Snowy Forest",
+  "ISLAND": "Island",
+  "LABORATORY": "Laboratory",
+  "END": "Final Destination",
```
Comment (bennybroseph) [inline src/locales/en/biome.ts:39]: Same goes for all, especially English. Please don't change english values. This also changes Space for some reason

**Review by bennybroseph — CHANGES_REQUESTED** [review]

File: src/data/biomes.ts [hunk 3]
```diff
@@ -6,22 +6,23 @@ import { TrainerType } from "./enums/trainer-type";
...
+    return i18next.t("biome:GRASS");
   case Biome.RUINS:
-    return "Ancient Ruins";
+    return i18next.t("biome:RUINS");
   case Biome.ABYSS:
-    return "The Abyss";
+    return i18next.t("biome:ABYSS");
   case Biome.END:
-    return "???";
+    return i18next.t("biome:END");
```
Comment (bennybroseph) [inline src/data/biomes.ts:23]: I'm pretty sure images are loaded based on the return value of this function. Just want to make sure images are loading for all biomes in a non-english language with this change

**Review by bennybroseph — CHANGES_REQUESTED** [review]

File: src/locales/de/biome.ts (hunk 1 above)
Comment (CodeTappert) [reply src/locales/de/biome.ts:39]: I just copied this from the other PR. Okay i will set it to ??? for key and value in all languages

File: src/data/biomes.ts (hunk 3 above)
Comment (CodeTappert) [reply src/data/biomes.ts:23]: I am do a full through with this german changes

**Comment by Adri1** [comment]
The work has been already done part of the -long awating- PR #918 , including a French translation already pushed into it

**Comment by CodeTappert** [comment]
> The work has been already done part of the -still waiting- PR #918 , including a bit more accurate French translation already pushed into it

918 Is pretty big. We should split it into smaller ones. Which i am doing with this

File: src/locales/en/biome.ts (hunk 2 above)
Comment (CodeTappert) [reply src/locales/en/biome.ts]: Changed End to ??? in all languages. 

I think the english values should be the same now as original (if not please tell me which or give me list of them)

File: src/data/biomes.ts (hunk 3 above)
Comment (CodeTappert) [reply src/data/biomes.ts:23]: Plains
- image
Sea
- image
Construction Site
- image


I hope this is enough proof of it not making problems with the images

**Comment by Adri1** [comment]
@CodeTappert I submitted you an updated version of the French locale to your banch

**Comment by CodeTappert** [comment]
> @CodeTappert I submitted you an updated version of the French locale to your banch

I will have sangara have a look at it. Since they did it. I dont know any french

File: src/locales/en/biome.ts (hunk 2 above)
Comment (CodeTappert) [reply src/locales/en/biome.ts]: I updated the english names accoridng to https://wiki.pokerogue.net/biomes:biomes

File: src/data/biomes.ts [hunk 4]
```diff
@@ -6,22 +6,23 @@ import { TrainerType } from "./enums/trainer-type";
...
+import i18next from "i18next";
 
 export function getBiomeName(biome: Biome | -1) {
   if (biome === -1) {
-    return "Somewhere you can't remember";
+    return i18next.t("biome:unknownLocation");
   }
   switch (biome) {
   case Biome.GRASS:
-    return "Grassy Field";
```
Comment (bennybroseph) [inline src/data/biomes.ts:16]: this was also changed for english

**Review by bennybroseph — CHANGES_REQUESTED** [review]

File: src/data/biomes.ts (hunk 4 above)
Comment (CodeTappert) [reply src/data/biomes.ts:16]: Well so the wiki is wrong?

Okay i will change it back to grassy field

**Review by bennybroseph — APPROVED** [review]

**Pull request merged**